docs/sources/reference/api/docker_remote_api_v1.17.md

```diff
@@ -802,6 +804,138 @@ Status Codes:
 -   **404** – no such container
 -   **500** – server error
 
+### Get files or folders from a container as a Tar Archive.
+
+`GET /containers/(id)/copy`
+
+Get an archive of a resource in the filesystem of container `id`.
+
+Query Parameters:
+
+-   **path** - The resource in the container's filesystem. If not an absolute
+        path, it is relative to the container's working directory. If path is
+        inside the mount point of a volume, the volume's data is archived. If
+        the path specifies a directory and a volume is mounted lower in that
+        directory hierarchy, the volume's contents are not included in the
+        archive. Required.
+
+**Example request**:
+
+        GET /containers/8cce319429b2/copy?path=/foo/ HTTP/1.1
+
+**Example response**:
+
+        HTTP/1.1 200 OK
+        Content-Type: application/x-tar
+
+        {{ TAR STREAM }}
+
+Status Codes:
+
+-   **200** – no error
+-   **400** - client error, details in JSON response body, one of:
+        - path cannot be empty (missing path query parameter)
+        - no such file or directory (path does not exist)
+        - not a directory (path ends with a separator but exists as a file)
+-   **404** – no such container
+-   **500** – server error
+
+### Upload files or folders to a container from a Tar Archive.
+
+`PUT /containers/(id)/copy`
+
+Upload an archive of a resource to be extracted to a path in the filesystem
+of container `id`.
+
+Query Parameters:
+
+-   **srcPath** - Specifies the original path from which the payload archive
+        was created. Used as a hint to the server when determining if the
+        contents of the archive are a single file or should be interpreted
+        as directory contents. If the first entry of the archive has a name
```

Comment:
But is this really an issue if we DO NOT create the target dir for them magically?  If we assume, like cp does, that if you want to copy into a dir then you must create the target dir before hand then I think the issue goes away, doesn't it?  Because then if the target doesn't exist they must be asking for a file to be created. And if more than one src is specified its an error.